### PR TITLE
feat(edit-ema): #26997 Add page properties functionality

### LINKED ISSUE #26997: Edit EMA: Allow Users to Edit Page Properties
### Parent
#26644 

### User Story
As a dotCMS user, I want to be able to edit the properties of a page in the new page builder, so I can make changes to the title, url, SEO, etc...

```[tasklist]
### Acceptance Criteria
- [x] User clicks in the `...` button and it shows the dialog with the jsp to edit the page contentlet
- [x] User can edit the contentlet and trigger any workflow actions assigned
- [x] The page editor should reload to update the state of the page
- [x] The user can change the url and it should be reflected in the editor state, including the nextjs iframe
```

### Proposed Objective
Core Features

### Proposed Priority
Priority 3 - Average

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.:
N/A

### Assumptions & Initiation Needs
- The JSP emits events
- PrimgNG dialog close events can be used

### Quality Assurance Notes & Workarounds
N/A

### PR DESCRIPTION
**Pull request by zJaaal**
### Proposed Changes
* Add a dialog to the shell and use the same logic as in editor.
* Adapt the store to handle the opening and closing of dialogs by context (editor or shell at the moment)
* Add test cases
* Send a timestamp in the iframeURL to trigger a reload when the queryParams do not update. This happens when we edit the page properties. We need to force a reload but we didn't changed any queryParam, so we send a timestamp as a flag to reload.

### Checklist
- [x] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Screenshots

https://github.com/dotCMS/core/assets/63567962/e359ca47-ae5b-499a-8bc7-9c1004fac5e2

### CODE REVIEW

File: core-web/libs/portlets/dot-ema/src/lib/dot-ema-shell/components/edit-ema-navigation-bar/edit-ema-navigation-bar.component.html [hunk 1]
```diff
@@ -1,9 +1,16 @@
 <nav class="edit-ema-nav-bar">
+    <ng-container *ngFor="let item of items">
+        <ng-container
+            [ngTemplateOutlet]="item.action ? button : anchor"
```
Comment (manuelrojas) [inline core-web/libs/portlets/dot-ema/src/lib/dot-ema-shell/components/edit-ema-navigation-bar/edit-ema-navigation-bar.component.html:4]: Cool

**Review by manuelrojas — APPROVED** [review]

File: core-web/libs/portlets/dot-ema/src/lib/dot-ema-shell/dot-ema-shell.component.ts [hunk 2]
```diff
@@ -133,4 +149,34 @@ export class DotEmaShellComponent implements OnInit, OnDestroy {
         this.destroy$.next(true);
         this.destroy$.complete();
     }
+
+    resetDialogIframeData() {
+        this.store.resetDialog();
+    }
+
+    onIframeLoad() {
+        this.store.setDialogIframeLoading(false);
```
Comment (hmoreras) [inline core-web/libs/portlets/dot-ema/src/lib/dot-ema-shell/dot-ema-shell.component.ts:158]: this also can be in a `tap` inside the `pipe`

**Review by hmoreras — APPROVED** [review]

File: core-web/libs/portlets/dot-ema/src/lib/dot-ema-shell/dot-ema-shell.component.spec.ts [hunk 3]
```diff
@@ -147,4 +150,91 @@ describe('DotEmaShellComponent', () => {
             expect(navigate).toHaveBeenCalledWith(['/pages']);
         });
     });
+
+    describe('dialog', () => {
```
Comment (fmontes) [inline core-web/libs/portlets/dot-ema/src/lib/dot-ema-shell/dot-ema-shell.component.spec.ts:154]:
```suggestion
    describe('page properties', () => {
```

File: core-web/libs/portlets/dot-ema/src/lib/dot-ema-shell/store/dot-ema.store.ts [hunk 4]
```diff
@@ -24,11 +24,13 @@ import { insertContentletInContainer } from '../../utils';
...
+
 export interface EditEmaState {
     editor: DotPageApiResponse;
     url: string;
     dialogIframeURL: string;
-    dialogVisible: boolean;
+    dialogContext: DialogContext;
     dialogHeader: string;
     dialogIframeLoading: boolean;
     dialogType: DialogType;
```
Comment (fmontes) [inline core-web/libs/portlets/dot-ema/src/lib/dot-ema-shell/store/dot-ema.store.ts:36]: `dialogContext` and `dialogType` nahhh keep it simple.

File: core-web/libs/portlets/dot-ema/src/lib/dot-ema-shell/dot-ema-shell.component.ts [hunk 5]
```diff
@@ -56,21 +60,25 @@ import { NavigationBarItem } from '../shared/models';
...
     private readonly route = inject(ActivatedRoute);
     private readonly router = inject(Router);
     private readonly store = inject(EditEmaStore);
     private readonly siteService = inject(SiteService);
 
     private readonly destroy$ = new Subject<boolean>();
+    private queryParams: DotPageApiParams;
     pageToolsVisible = false;
 
+    dialogState$ = this.store.dialogState$;
```
Comment (fmontes) [inline core-web/libs/portlets/dot-ema/src/lib/dot-ema-shell/dot-ema-shell.component.ts:73]: you can use `dialogType` for this selector...

File: core-web/libs/portlets/dot-ema/src/lib/dot-ema-shell/dot-ema-shell.component.html [hunk 6]
```diff
@@ -6,3 +6,29 @@
...
+    #dialog
+    *ngIf="dialogState$ | async as ds"
+    [visible]="ds.context === 'shell'"
+    [style]="{ height: '90vh', width: '90vw' }"
+    [header]="ds.header"
+    [draggable]="false"
+    [resizable]="false"
+    [maximizable]="true"
+    [modal]="true"
+    (visibleChange)="resetDialogIframeData()"
```
Comment (fmontes) [inline core-web/libs/portlets/dot-ema/src/lib/dot-ema-shell/dot-ema-shell.component.html:19]:
```suggestion
    (visibleChange)="store.resetDialog()"
```

File: core-web/libs/portlets/dot-ema/src/lib/dot-ema-shell/store/dot-ema.store.spec.ts [hunk 7]
```diff
@@ -130,20 +134,21 @@ describe('EditEmaStore', () => {
...
                     editor: mockResponse,
                     url: 'test-url',
                     dialogIframeURL:
-                        '/html/ng-contentlet-selector.jsp?ng=true&container_id=123&add=test&language_id=1',
+                        '/html/ng-contentlet-selector.jsp?ng=true&container_id=1234&add=test&language_id=1',
                     dialogIframeLoading: true,
                     dialogHeader: 'Search Content',
-                    dialogVisible: true,
+                    dialogContext: 'editor',
                     dialogType: 'content'
```
Comment (fmontes) [inline core-web/libs/portlets/dot-ema/src/lib/dot-ema-shell/store/dot-ema.store.spec.ts:152]: 👎🏽

**Review by fmontes — CHANGES_REQUESTED** [review]

File: core-web/libs/portlets/dot-ema/src/lib/dot-ema-shell/dot-ema-shell.component.ts (hunk 2 above)
Comment (zJaaal) [reply core-web/libs/portlets/dot-ema/src/lib/dot-ema-shell/dot-ema-shell.component.ts:158]: I prefer to call it explicitly

**Review by KevinDavilaDotCMS — APPROVED** [review]

**Review by fmontes — APPROVED** [review]

**Review by oidacra — APPROVED** [review]

**Pull request merged**